src/controller/Controller.py

```diff
@@ -50,6 +50,47 @@ def get_generator(self) -> str:
         """
         return self._gen
 
+    def sample_random_input(
+        self, n: int, attr_sample_mode: Union[int, str]
+    ) -> Dict[str, np.ndarray]:
+        """
+        Samples input from a uniform distribution.
+
+        Args:
+            n (int): Number of samples to generate native inputs for.
+            attr_sample_mode (Union[int, str]): A integer if number of attributes
+                to use from `self._attrs` should be `attr_sample_mode`,
```

Comment:
too technical, do not expose private members in docstrings

maybe a bit confusing to refer to itself in third-person

Perhaps: `The number of random attributes to choose, or "random"...`.

If you want to mention where the attributes may be found, either refer to the construction, or maybe expose the attributes through a public getter which you may refer to.